mungegithub/mungers/approval-handler.go

```diff
@@ -109,6 +118,60 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 
 }
 
+func notificationNeedsUpdate(obj *github.MungeObject) bool {
+	notificationMatcher := mungeComment.MungerNotificationName(approvalNotificationName)
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error("Could not list the comments for PR%v", obj.Issue.Number)
+		return false
+	}
+	latestNotification := mungeComment.FilterComments(comments, notificationMatcher).GetLast()
+	latestApprove := mungeComment.FilterComments(comments, mungeComment.CommandName("approve")).GetLast()
+	if latestNotification == nil {
+		return true
+	}
+	if latestApprove == nil {
+		// there was already a bot notitication and nothing has changed since
+		return false
+	}
+	if latestApprove.CreatedAt.After(*latestNotification.CreatedAt) {
+		// there has been approval since last notification
+		obj.DeleteComment(latestNotification)
+		return true
+	}
+	lastModified := obj.LastModifiedTime()
+	if latestApprove.CreatedAt.Before(*lastModified) {
+		obj.DeleteComment(latestNotification)
+		return true
+	}
+	return false
+}
+
+func createMessage(obj *github.MungeObject, filesNeedApproval sets.String) {
+	files, err := obj.ListFiles()
+	if err != nil {
+		glog.Error("Could not list the files for PR%v", obj.Issue.Number)
+		return
+	}
+	approvedFiles := sets.String{}
+	for _, fn := range files {
+		if !filesNeedApproval.Has(*fn.Filename) {
+			approvedFiles.Insert(*fn.Filename)
+		}
+	}
+	context := bytes.Buffer{}
+	context.WriteString("The following files have been approved:\n")
+	fileFmt := "\t%s"
+	for _, fn := range approvedFiles {
+		context.WriteString(fmt.Sprintf(fileFmt, fn))
```

Comment:
should be the `OWNERS` file, not the file per se. Or we will have too many of these. And it'd be nice if it's a link. and if we can see who approved what ;-)